### PR TITLE
JENKINS-43980: add support for expanding variables in IssueFieldUpdaterStep

### PR DESCRIPTION
**Pull request by warden**
- replaces #162
- bumped Jenkins version to 2.249.3 for EnvVars to be available in workflow steps
- added a test for variable expansion

### CODE REVIEW

**Comment by warden** [comment]
Looking good, I'll just need to do a manual test and we'll be good to go.

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <jira-rest-client.version>5.2.2</jira-rest-client.version>
     <fugue.version>3.0.0</fugue.version>
     <!-- jenkins -->
-    <jenkins.version>2.222.4</jenkins.version>
+    <jenkins.version>2.249.3</jenkins.version>
```
Comment (olamy) [inline pom.xml:26]: This should be part of a separate PR to have his own line in the release notes

**Review by olamy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (warden) [reply pom.xml:26]: I did it in a separate commit, I think that's enough, for the release notes, we can just edit it manually and link the commit id.

Comment (olamy) [reply pom.xml:26]: We have tool to avoid manual errors/forgetting btw someone else can cut the release and forget.
but that's fine as long as you do not forget it that will be fine :)

Comment (olamy) [reply pom.xml:26]: BTW this branch has 6 commits. will you squash 5 and 1 (with jenkins version upgrade)?

Comment (warden) [reply pom.xml:26]: The previous commits need to stay (as they were picked up from the previous PR that will be auto-closed when this PR gets merged). 
Also as I understand, the merge commits cannot be squashed, they can be rebased though but we had a discussion in the past about it with jglick that's irrelevant as they shouldn't be shown after merge.. but I might be wrong here :)

Comment (olamy) [reply pom.xml:26]: Ideally we should have 1 commit (PR squashed) per feature. 
So here we should have 2 commits:
- the new feature (co-authored by you and the other contributor)
- jenkins core version required upgrade

perso I prefer to have clean PR so I use rebase only to not have those noisy merge commits.

Comment (olamy) [reply pom.xml:26]: at least update the title of the PR to have some details on core upgrade

Comment (warden) [reply pom.xml:26]: Done the squashing, looks fine to me now. 
I'll mention the core upgrade in a separate bullet point in the release notes referring to the commit id.

**Review by olamy — APPROVED** [review]
awesome job thanks!
